plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -75,8 +75,9 @@ class ProgramConverter(val session: FirSession, override val config: PluginConfi
     override fun embedFunction(symbol: FirFunctionSymbol<*>): FunctionEmbedding =
         methods.getOrPut(symbol.embedName(this)) {
             val signature = embedFullSignature(symbol)
-            val embedding = UserFunctionEmbedding(processCallable(symbol, signature))
-            embedding.viperMethod = convertMethodWithoutBody(symbol, signature)
+            val callable = processCallable(symbol, signature)
+            val embedding = UserFunctionEmbedding(callable)
+            embedding.viperMethod = callable.toViperMethod()
```

Comment:
Wouldn't be possible initializing the field `viperMethod` within an `init` block in `UserFunctionEmbedding` class?

Reply:
This is not the only place `viperMethod` is set; `UserFunctionEmbedding` does not know whether the method should be set based on the `callable` passed in or not.